File: solution/78/solution.cpp
#include <stdio.h>
#include <vector>

class Solution {
public:
    void subsetstool(int i, std::vector<int>& nums, std::vector<int>& currentarray, std::vector<vector<int>>& result){
        if(i>=nums.size())return;
        currentarray.push_back(nums[i]);
        result.push_back(currentarray);
        subsetstool(i+1, nums, currentarray, result);
        currentarray.pop_back();
        subsetstool(i+1, nums, currentarray, result);
    }
    std::vector<vector<int>> subsets(std::vector<int>& nums) {
        std::vector<vector<int>> result;
        std::vector<int> currentarray;
        result.push_back(currentarray);
        subsetstool(0, nums, currentarray, result);
        return result;
    }
};

File: solution/78/solution.py
import copy
class Solution(object):
    def generate(self, n, nums, item, result):
        if n >= len(nums):return
        result.append(item+[nums[n]])
        item += [nums[n]]
        self.generate(n+1, nums, item, result)
        item.pop()
        self.generate(n+1, nums, item, result)

    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        result = [[]]
        self.generate(0, nums, [], result)
        return result

s = Solution()
print s.subsets([1,2,3])